BitwardenShared/Core/Auth/Services/API/Account/Requests/VerifyEmailTokenRequest.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+import Networking
+
+// MARK: - VerifyEmailTokenRequestError
+
+/// Errors that can occur when sending a `VerifyEmailTokenRequest`.
+enum VerifyEmailTokenRequestError: Error, Equatable {
+    /// The token provided by email is expired or user is already used.
+    ///
+    case tokenExpired
+}
+
+// MARK: - VerificationEmailClickedRequest
+
+/// The API request sent when verifying the token received by email.
+///
+struct VerifyEmailTokenRequest: Request {
+    typealias Response = EmptyResponse
+
+    typealias Body = VerifyEmailTokenRequestModel
+
+    /// The body of this request.
+    var body: VerifyEmailTokenRequestModel? {
+        requestModel
+    }
+
+    /// The HTTP method for this request.
+    let method: HTTPMethod = .post
+
+    /// The URL path for this request.
+    var path: String = "/accounts/register/verification-email-clicked"
+
+    /// The request details to include in the body of the request.
+    let requestModel: VerifyEmailTokenRequestModel
+
+    // MARK: Methods
+
+    func validate(_ response: HTTPResponse) throws {
+        switch response.statusCode {
+        case 400 ..< 500:
+            guard let errorResponse = try? ErrorResponseModel(response: response) else { return }
+
+            if errorResponse.message.contains("Expired link") {
+                throw VerifyEmailTokenRequestError.tokenExpired
+            }
```

Comment:
🤔 Does the server send a specific error number or something else we could use instead of the message string? I think this kind of error message comparison is brittle and may be a bit harder to know what's happening if there's an issue because the message changed.

Reply:
I did check it but not: [source](https://github.com/bitwarden/server/pull/4520/files#diff-d23fef2a55cc8602622925f7dca275305672a576384e77b1d1e616282e1b6999R133)

Reply:
Alright I'll raise it up, thanks for the info! 😄 I'll approve it as is now.